Description: Remove overreaching citation notice
 Quoting the gpl-faq:
 .
 > Does the GPL allow me to add terms that would require citation or
 > acknowledgment in research papers which use the GPL-covered software or its
 > output? (#RequireCitation)
 > .
 > No, this is not permitted under the terms of the GPL. While we recognize
 > that proper citation is an important part of academic publications,
 > citation cannot be added as an additional requirement to the GPL. Requiring
 > citation in research papers which made use of GPL'd software goes beyond
 > what would be an acceptable additional requirement under section 7(b) of
 > GPLv3, and therefore would be considered an additional restriction under
 > Section 7 of the GPL. And copyright law does not allow you to place such a
 > requirement on the output of software, regardless of whether it is licensed
 > under the terms of the GPL or some other license.
 .
 Therefore, removing this to make parallel GPL-compliant.

 While I agree with the GPL FAQ I found the following change inside the
 code while refreshing the patch:
 .
     # Before changing this line, please read
     # https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 .
 Thus this patch is again up for discussion.

Comment: Upstream has enforced his statement by adding
    # You accept to be put in a public hall of shame by removing
    # these lines.
 before the code lines we change in a patch.  Andreas Tille does not think that this
 kind of statements invalidate the right to change the code as it is granted by GPL
 and was discussed above.

Author: Didier Raboud <odyx@debian.org>
        Ian Turner <vectro@vectro.org> (for the documentation parts)
Bug-Debian: https://bugs.debian.org/905674

Last-Update:  Andreas Tille  Wed, 27 Jul 2022 09:27:03 +0200

--- parallel.orig/src/env_parallel.dash
+++ parallel/src/env_parallel.dash
@@ -611,7 +611,7 @@
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- parallel.orig/src/env_parallel.ksh
+++ parallel/src/env_parallel.ksh
@@ -611,7 +611,7 @@
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- parallel.orig/src/env_parallel.pod
+++ parallel/src/env_parallel.pod
@@ -807,9 +807,6 @@
 O. Tange (2018): GNU Parallel 2018, March 2018, ISBN 9781387509881,
 DOI: 10.5281/zenodo.1146014.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- parallel.orig/src/env_parallel.sh
+++ parallel/src/env_parallel.sh
@@ -611,7 +611,7 @@
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- parallel.orig/src/env_parallel.zsh
+++ parallel/src/env_parallel.zsh
@@ -611,7 +611,7 @@
 	echo "Web site: https://www.gnu.org/software/parallel"
 	echo
 	echo "When using programs that use GNU Parallel to process data for publication"
-	echo "please cite as described in 'parallel --citation'."
+	echo "please cite as described in the manpage."
 	echo
 	return 255
     fi
--- parallel.orig/src/parallel
+++ parallel/src/parallel
@@ -1977,11 +1977,11 @@
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 	 # You accept to be put in a public hall-of-shame by removing
 	 # these lines
-	 ("bibtex|citation".
-	  "[Print the citation notice and BibTeX entry for GNU parallel, ".
-	  "silence citation notice for all future runs, and exit. ".
-	  "It will not run any commands]"
-	  => \$opt::citation),
+#	 ("bibtex|citation".
+#	  "[Print the citation notice and BibTeX entry for GNU parallel, ".
+#	  "silence citation notice for all future runs, and exit. ".
+#	  "It will not run any commands]"
+#	  => \$opt::citation),
 	 "will-cite|willcite|nn|nonotice|no-notice" => \$opt::willcite,
 	 # Termination and retries
 	 ('halt-on-error|haltonerror|halt=s'.
@@ -2221,10 +2221,10 @@
     # https://www.gnu.org/software/parallel/parallel_design.html#citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
     # You accept to be added to a public hall-of-shame by removing the lines
-    if(defined $opt::citation) {
-	citation(\@argv_before,\@ARGV);
-	wait_and_exit(0);
-    }
+#    if(defined $opt::citation) {
+#	citation(\@argv_before,\@ARGV);
+#	wait_and_exit(0);
+#    }
     # no-* overrides *
     if($opt::nokeeporder) { $opt::keeporder = undef; }
 
@@ -2618,7 +2618,7 @@
     #
     # If you want GNU Parallel to be maintained in the future, you
     # should keep this line:
-    citation_notice();
+#    citation_notice();
     # This is because _YOU_ actively make it harder to justify
     # spending time developing GNU Parallel by removing it.
 
@@ -5811,9 +5811,9 @@
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 	 # You accept to be put in a  public hall of shame  by removing
 	 # these lines
-	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
-	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
-	 "",
+#	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
+#	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
+#	 "",
 	 "",);
 }
 
@@ -5951,7 +5951,7 @@
 	 "",
 	 "Web site: https://www.gnu.org/software/${Global::progname}\n",
 	 "When using programs that use GNU Parallel to process data for publication",
-	 "please cite as described in 'parallel --citation'.\n",
+	 "please cite as described in the manpage.\n",
 	);
 }
 
--- parallel.orig/src/parallel.pod
+++ parallel/src/parallel.pod
@@ -679,31 +679,31 @@
 
 # You accept to be added to a public hall of shame by
 # removing this section.
-=item B<--bibtex>
-
-=item B<--citation>
-
-Print the citation notice and BibTeX entry for GNU B<parallel>,
-silence citation notice for all future runs, and exit. It will not run
-any commands.
-
-If it is impossible for you to run B<--citation> you can instead use
-B<--will-cite>, which will run commands, but which will only silence
-the citation notice for this single run.
-
-If you use B<--will-cite> in scripts to be run by others you are
-making it harder for others to see the citation notice.  The
-development of GNU B<parallel> is indirectly financed through
-citations, so if your users do not know they should cite then you are
-making it harder to finance development. However, if you pay 10000
-EUR, you have done your part to finance future development and should
-feel free to use B<--will-cite> in scripts.
-
-If you do not want to help financing future development by letting
-other users see the citation notice or by paying, then please consider
-using another tool instead of GNU B<parallel>. You can find some of
-the alternatives in B<man parallel_alternatives>.
-
+#=item B<--bibtex>
+#
+#=item B<--citation>
+#
+#Print the citation notice and BibTeX entry for GNU B<parallel>,
+#silence citation notice for all future runs, and exit. It will not run
+#any commands.
+#
+#If it is impossible for you to run B<--citation> you can instead use
+#B<--will-cite>, which will run commands, but which will only silence
+#the citation notice for this single run.
+#
+#If you use B<--will-cite> in scripts to be run by others you are
+#making it harder for others to see the citation notice.  The
+#development of GNU B<parallel> is indirectly financed through
+#citations, so if your users do not know they should cite then you are
+#making it harder to finance development. However, if you pay 10000
+#EUR, you have done your part to finance future development and should
+#feel free to use B<--will-cite> in scripts.
+#
+#If you do not want to help financing future development by letting
+#other users see the citation notice or by paying, then please consider
+#using another tool instead of GNU B<parallel>. You can find some of
+#the alternatives in B<man parallel_alternatives>.
+#
 
 =item B<--block> I<size>
 
@@ -4475,9 +4475,6 @@
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- parallel.orig/src/parallel_alternatives.pod
+++ parallel/src/parallel_alternatives.pod
@@ -4627,9 +4627,6 @@
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- parallel.orig/src/parallel_design.pod
+++ parallel/src/parallel_design.pod
@@ -1416,18 +1416,6 @@
 cost work and which have not been tested are, however, unlikely to be
 prioritized.
 
-Running B<parallel --citation> one single time takes less than 10
-seconds, and will silence the citation notice for future runs. This is
-comparable to graphical tools where you have to click a checkbox
-saying "Do not show this again". But if that is too much trouble for
-you, why not use one of the alternatives instead?  See a list in:
-B<man parallel_alternatives>.
-
-As the request for citation is not a legal requirement this is
-acceptable under GPLv3 and cleared with Richard M. Stallman
-himself. Thus it does not fall under this:
-https://www.gnu.org/licenses/gpl-faq.en.html#RequireCitation
-
 
 =head1 Ideas for new design
 
--- parallel.orig/src/parallel_tutorial.pod
+++ parallel/src/parallel_tutorial.pod
@@ -2998,9 +2998,6 @@
   
     O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
     ;login: The USENIX Magazine, February 2011:42-47.
-  
-  This helps funding further development; AND IT WON'T COST YOU A CENT.
-  If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
 
 When asking for help, always report the full output of this:
 
@@ -3018,7 +3015,7 @@
   Web site: https://www.gnu.org/software/parallel
   
   When using programs that use GNU Parallel to process data for publication
-  please cite as described in 'parallel --citation'.
+  please cite as described in the manpage.
 
 In scripts B<--minversion> can be used to ensure the user has at least
 this version:
@@ -3031,38 +3028,6 @@
   20160322
   Your version is at least 20130722.
 
-If you are using GNU B<parallel> for research the BibTeX citation can be
-generated using B<--citation>:
-
-  parallel --citation
-
-Output:
-
-  Academic tradition requires you to cite works you base your article on.
-  When using programs that use GNU Parallel to process data for publication
-  please cite:
-  
-  @article{Tange2011a,
-    title = {GNU Parallel - The Command-Line Power Tool},
-    author = {O. Tange},
-    address = {Frederiksberg, Denmark},
-    journal = {;login: The USENIX Magazine},
-    month = {Feb},
-    number = {1},
-    volume = {36},
-    url = {https://www.gnu.org/s/parallel},
-    year = {2011},
-    pages = {42-47},
-    doi = {10.5281/zenodo.16303}
-  }
-  
-  (Feel free to use \nocite{Tange2011a})
-  
-  This helps funding further development; AND IT WON'T COST YOU A CENT.
-  If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-  
-  If you send a copy of your published article to tange@gnu.org, it will be
-  mentioned in the release notes of next version of GNU Parallel.
   
 With B<--max-line-length-allowed> GNU B<parallel> will report the maximal
 size of the command line:
@@ -3155,7 +3120,7 @@
 
 =item *
 
-Please cite GNU B<parallel> in you publications (use B<--citation>)
+Please cite GNU B<parallel> in you publications
 
 =back
 
--- parallel.orig/src/parset.pod
+++ parallel/src/parset.pod
@@ -195,9 +195,6 @@
 O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login:
 The USENIX Magazine, February 2011:42-47.
 
-This helps funding further development; and it won't cost you a cent.
-If you pay 10000 EUR you should feel free to use GNU Parallel without citing.
-
 Copyright (C) 2007-10-18 Ole Tange, http://ole.tange.dk
 
 Copyright (C) 2008-2010 Ole Tange, http://ole.tange.dk
--- parallel.orig/src/sem
+++ parallel/src/sem
@@ -1977,11 +1977,11 @@
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 	 # You accept to be put in a public hall-of-shame by removing
 	 # these lines
-	 ("bibtex|citation".
-	  "[Print the citation notice and BibTeX entry for GNU parallel, ".
-	  "silence citation notice for all future runs, and exit. ".
-	  "It will not run any commands]"
-	  => \$opt::citation),
+#	 ("bibtex|citation".
+#	  "[Print the citation notice and BibTeX entry for GNU parallel, ".
+#	  "silence citation notice for all future runs, and exit. ".
+#	  "It will not run any commands]"
+#	  => \$opt::citation),
 	 "will-cite|willcite|nn|nonotice|no-notice" => \$opt::willcite,
 	 # Termination and retries
 	 ('halt-on-error|haltonerror|halt=s'.
@@ -2221,10 +2221,10 @@
     # https://www.gnu.org/software/parallel/parallel_design.html#citation-notice
     # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
     # You accept to be added to a public hall-of-shame by removing the lines
-    if(defined $opt::citation) {
-	citation(\@argv_before,\@ARGV);
-	wait_and_exit(0);
-    }
+#    if(defined $opt::citation) {
+#	citation(\@argv_before,\@ARGV);
+#	wait_and_exit(0);
+#    }
     # no-* overrides *
     if($opt::nokeeporder) { $opt::keeporder = undef; }
 
@@ -2618,7 +2618,7 @@
     #
     # If you want GNU Parallel to be maintained in the future, you
     # should keep this line:
-    citation_notice();
+#    citation_notice();
     # This is because _YOU_ actively make it harder to justify
     # spending time developing GNU Parallel by removing it.
 
@@ -5811,9 +5811,9 @@
 	 # https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt
 	 # You accept to be put in a  public hall of shame  by removing
 	 # these lines
-	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
-	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
-	 "",
+#	 "This helps funding further development; AND IT WON'T COST YOU A CENT.",
+#	 "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.",
+#	 "",
 	 "",);
 }
 
@@ -5951,7 +5951,7 @@
 	 "",
 	 "Web site: https://www.gnu.org/software/${Global::progname}\n",
 	 "When using programs that use GNU Parallel to process data for publication",
-	 "please cite as described in 'parallel --citation'.\n",
+	 "please cite as described in the manpage.\n",
 	);
 }
 
